decompiler/frontend/binaryninja/handlers/globals.py

```diff
@@ -12,30 +23,161 @@ class GlobalHandler(Handler):
     # Dict translating endianness between the binaryninja enum and pythons literals
     Endian = {Endianness.LittleEndian: "little", Endianness.BigEndian: "big"}
 
+    def __init__(self, lifter):
+        super().__init__(lifter)
+        self._datavariable_types = {
```

Comment:
I think the way you define and use this dictionary in code is very confusing, maybe because of the naming.
The dictionary looks like something for lifting types (`lift_..._type`), but  the result of lifting are various expression instances and not types as function names suggest. This makes the code very hard to comprehend.

Reply:
I see what you mean, but actually i use the type of the datavariable to lift them.
Different datavariable types (pointer, basic type, void type and constant type) yield different methods. 

Maybe change the name to `_lift_datavariable_type` and add type hints will help to understand the code?

Reply:
What about `_lift_datavarible_by_type` or something like that? This way it would be clear, what is being lifted (datavariable, not its type)?

Reply:
Sure and yes lift datavariable __by__ type